java/src/jmri/jmrix/roco/z21/Z21ReporterManager.java

```diff
@@ -153,6 +154,59 @@ public void enableInternalReporterCreationFromMessages() {
         autoCreateInternalReporter = true;
     }
 
+    private String createSystemName(String curAddress, String prefix) throws JmriException {
+        int encoderAddress = 0;
+        int input = 0;
+
+        if (curAddress.contains(":")) {
+            // Address format passed is in the form of encoderAddress:input or T:turnout address
+            int seperator = curAddress.indexOf(":");
+            input = Integer.parseInt(curAddress.substring(seperator + 1));
+            try {
+                encoderAddress = Integer.parseInt(curAddress.substring(0, seperator));
+                return getSystemPrefix() + typeLetter() + encoderAddress
+                                         + ":" + input;
+            } catch (NumberFormatException ex) {
+                // system name may include hex values for CAN sensors.
+                try {
+                    encoderAddress = Integer.parseInt(curAddress.substring(0, seperator), 16);
+                    return getSystemPrefix() + typeLetter() + String.format("%4x", encoderAddress) + ":" + input;
+                } catch (NumberFormatException ex1) {
+                    log.error("Unable to convert {} into the cab and input format of nn:xx", curAddress);
+                    throw new JmriException("Hardware Address passed should be a number");
+                }
+            }
+        } else {
+            // Entered in using the old format
+            try {
+                int iName = Integer.parseInt(curAddress);
+                return getSystemPrefix() + typeLetter() + iName;
+            } catch (NumberFormatException ex) {
+                log.error("Unable to convert {} Hardware Address to a number", curAddress);
+                throw new JmriException("Hardware Address passed should be a number");
+            }
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Reporter getBySystemName(String sName){
+       Reporter r = super.getBySystemName(sName);
+       if(r == null && sName.contains(":")) {
+          // normalize the hex characters in the system name to upper case.
+          String curAddress = sName.substring(getSystemPrefix().length() +1);
+          try {
+             return super.getBySystemName(createSystemName(curAddress, getSystemPrefix()));
+          } catch (JmriException je) {
+             // format isn't correct, but s already equals null, so just return
```

Comment:
`but s already equals null` should be `but r already equals null`

The letter `s` should be replaced with the letter `r`.